include/motor-control/core/motor_interrupt_handler.hpp

```diff
@@ -37,45 +36,135 @@ namespace motor_handler {
+        if (overflow(old_position, position_tracker) == true) {
+            buffered_move.target_position = old_position;
+            position_tracker = old_position;
+            return false;
+        }
+        return bool((old_position ^ position_tracker) & tick_flag);
+    }
 
     void update_move() {
-        finish_current_move();
-        get_move(&buffered_move);
+        has_active_move = queue->try_read_isr(&buffered_move);
+        buffered_move.velocity += steps_per_tick;
+        // (TODO: lc) We should check the direction (and set respectively)
+        // the direction pin for the motor once a move is being pulled off the
+        // queue stack. We'll probably want to think about moving the hardware
+        // pin configurations out of motion controller.
     }
 
     void finish_current_move() {
         has_active_move = false;
-        buffered_move = Move{};
-        inc = 0x0;
+        set_buffered_move(Move{});
     }
 
-    void increment_counter() { inc++; }
+    bool pulse() {
+        /*
+         * Function to determine whether a motor step-line should
+         * be pulsed or not. It should return true if the step-line is to be
+         * pulsed, and false otherwise.
+         *
+         * An active move is true when there is a move that has been
+         * pulled from the queue. False otherwise.
+         *
+         * Logic:
+         * 1. If there is not currently an active move, we should check if there
+         * are any available on the queue.
+         * 2. If there is an active move, and stepping is possible, then we
+         * should increment the step counter and return true.
+         * 3. Finally, if there is an active move, but you can no longer step
+         * then active move should be set to false.
+         *
+         * This function is called from a timer interrupt. See `step_motor.cpp`.
+         */
+        if (!has_active_move && has_messages()) {
+            update_move();
+            return false;
+        } else if (has_active_move && can_step() && tick()) {
+            return true;
+        } else if (has_active_move && !can_step()) {
+            finish_current_move();
+            return false;
+        }
+        return false;
+    }
 
     void reset() {
+        /*
+         * Reset the position and all queued moves to the motor interrupt
+         * handler.
+         */
         queue->reset();
-        inc = 0x0;
+        position_tracker = 0x0;
         has_active_move = false;
     }
 
+    bool overflow(sq32_31 current, sq32_31 future) {
+        /*
+         * Check whether the position has overflowed. Return true if this has
+         * happened, and false otherwise.
+         *
+         * (TODO lc): If an overflow has happened, we should probably notify the
+         * server somehow that the position needed to be capped.
+         * Note: could not get the built-ins to work as expected.
+         */
+        return bool((current ^ future) & overflow_flag);
```

Comment:
so this is us implicitly limiting this to a uint63? honestly I think that's fine and perhaps better but we should be up front about it